server/server.go

```diff
@@ -2187,7 +2189,7 @@ func findBestQueueSub(sl []*subState) (rsub *subState) {
 
 		// Favor non stalled subscribers and clients that do not have
 		// failed heartbeats
-		if (!sStalled || rStalled) && (!sHasFailedHB || rHasFailedHB) && (sOut < rOut) {
+		if (!sStalled && !sHasFailedHB && (rStalled || rHasFailedHB)) || (sOut < rOut) {
```

Comment:
Yes. The only difference in behavior is that with the current one, we at least pick one (even if it is stalled) as long as the array has 1 element. In the above approach (as-is), we would potentially return `nil`. (just so you know, I am working on a change for better handling `stalled` semantic, especially for queue groups, which then could address the fact that we return `nil` here if all are stalled).